### PR TITLE
Add vectorized year

### PR DESCRIPTION
**Pull request by FateTHarlaown**
Signed-off-by: FateTHarlaown <623239185@qq.com>
PCP #5751

###  What have you changed?
Add vectorized year

###  What is the type of the changes?

Pick one of the following and delete the others:
- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?
Please select the tests that you ran to verify your changes:
- Unit test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?
No

###  Does this PR affect `tidb-ansible`?
No

If there is a configuration ###  Refer to a related PR or issue link (optional)
#5751

### CODE REVIEW

**Comment by FateTHarlaown** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/87

File: components/tidb_query/src/rpn_expr/impl_time.rs [hunk 1]
```diff
@@ -51,6 +52,24 @@ pub fn week_day(ctx: &mut EvalContext, t: &Option<DateTime>) -> Result<Option<In
...
+        Some(v) => v,
+        _ => return Ok(None),
+    };
+
+    if t.is_zero() {
+        if ctx.cfg.sql_mode.contains(SqlMode::NO_ZERO_DATE) {
+            return ctx
+                .handle_invalid_time_error(Error::incorrect_datetime_value(&format!("{}", t)))
+                .map(|_| Ok(None))?;
+        }
```
Comment (breeswish) [inline components/tidb_query/src/rpn_expr/impl_time.rs]: else?

**Comment by breeswish** [comment]
@hicqu PTAL, thanks!

**Comment by FateTHarlaown** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/87

**Comment by FateTHarlaown** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/87

**Comment by FateTHarlaown** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/87

**Pull request merged**